homeassistant/components/samsungtv/__init__.py

```diff
@@ -117,6 +118,19 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     # Initialize bridge
     bridge = await _async_create_bridge_with_updated_data(hass, entry)
 
+    # Ensure new token gets saved against the config_entry
+    def _update_token() -> None:
+        """Update config entry with the new token."""
+        hass.config_entries.async_update_entry(
+            entry, data={**entry.data, CONF_TOKEN: bridge.token}
+        )
+
+    def new_token_callback() -> None:
+        """Update config entry with the new token."""
+        run_callback_threadsafe(hass.loop, _update_token)
```

Comment:
We can use `hass.add_job` instead of `run_callback_threadsafe`, if we don't need to wait for the result.